toolkit/tools/grapher/grapher.go

```diff
@@ -36,6 +45,46 @@ func main() {
 	app.Version(exe.ToolkitVersion)
 	kingpin.MustParse(app.Parse(os.Args[1:]))
 
+	// CPU Profiling
```

Comment:
I was thinking about this again, could we add a new `profiler` type, and have a "constructor" and "destructor" for it in the exe package?
Something like
```go
type Profiler stuct {
	cpf, mpf, tf *os.File	 
}

func CreateProfiler(cpfFile, mpfFile, tfFile string) p *Profiler {
	p = &Profiler{}
	if cpfFile != "" {
		p.cpf, err = os.Create(cpfFile)
		// TODO: error check and start
	}
	if mpfFile != "" {
		p.mpf, err := os.Create(mpfFile)
		// TODO: error check and start
	}
	// etc...
	return p
}

func (p* Profiler) StopProfiler {
	if p.cpf != nil {
                pprof.StopCPUProfile()
		p.cpf.Close()
	}
        // etc...
}
```
Then we could call it like:
```go
profiler, err := exe.CreateProfiler(*cpuProfFile, *memProfFile, *traceFile)
// ...
defer profiler.StopProfiler()
```

Reply:
(I didn't actually compile any of this... so ymmv but the idea is there)

Reply:
Great idea, let me try to implement it this way.